doomed_patient/doomed_patient/Player.cpp

```diff
@@ -5,11 +5,21 @@
 
 Player::Player(PatientGame* game, VectorXY startCoordinates)
 	: Character(game, startCoordinates, game->getPlayerSprite()),
-	speed(2)
+	speed(0)
 {
+	currentCell = game->level.getCell(VectorXY(0, 0));
+	centre = currentCell->getCentre();
 }
 
 void Player::move(Directions::Direction direction)
 {
-	centre += Directions::getDirectionVector(direction) * speed;
+	
+	// Put collision detection here
+
+	if (currentCell->getEdge(direction)->isWall())
+	{
+		return;
+	}
+	currentCell = game->level.getCell(currentCell->getCoordinates() + Directions::getDirectionVector(direction));
+	centre = currentCell->getCentre();
```

Comment:
To make the player not move through walls in this way (where you're checking the edge of the current cell), you'll have to do a similar thing to what I did in the NPC class - it should make sure it's in the centre of the current cell before making the player unable to move